akd/tools/reranker.py

```diff
@@ -153,9 +156,428 @@ def __str__(self) -> str:
+                    results[criterion.name] = (0.0, "Error", "No score returned")
+
+            return results
+
+        except Exception as e:
+            logger.error(f"Error scoring result for all criteria: {e}")
+            # Return zero scores for all criteria on error
+            return {criterion.name: (0.0, "Error", str(e)) for criterion in self.config.scoring_criteria}
+
+    async def _rerank_results(self, query: str, results: list[SearchResultItem]) -> list[SearchResultItem]:
+        """
+        Rerank results by scoring each one against all criteria in a single LLM call per result.
+
+        Args:
+            query: The search query
+            results: List of results to rerank
+
+        Returns:
+            List of results sorted by weighted score
+        """
+        scored_results = []
+
+        for idx, result in enumerate(results):
+            result_content = self._extract_result_fields(result)
+
+            criterion_results = await self._score_result_all_criteria(
+                query=query,
+                result_content=result_content,
+            )
+
+            criterion_scores = {}
+            total_score = 0.0
+
+            for criterion in self.config.scoring_criteria:
+                numeric_score, category, reasoning = criterion_results.get(
+                    criterion.name,
+                    (0.0, "Error", "No score returned"),
+                )
+
+                print(
+                    f"Criterion: {criterion.name}, Score: {numeric_score}, Category: {category}, Reasoning: {reasoning}, weight: {criterion.weight}",
+                )
+
+                criterion_scores[criterion.name] = {
+                    "category": category,
+                    "score": numeric_score,
+                    "reasoning": reasoning,
+                    "weight": criterion.weight,
+                }
+
+                total_score += numeric_score * criterion.weight
+
+            result.score = total_score
+            result.extra["llm_reranker"] = {
+                "total_score": total_score,
+                "criterion_scores": criterion_scores,
+            }
+
+            if self.config.log_scores:
+                logger.info(
+                    f"Result {idx + 1}/{len(results)} | Total Score: {total_score:.3f} | "
+                    f"Scores: {', '.join(f'{k}={v["score"]:.1f}' for k, v in criterion_scores.items())}",
+                )
+
+            scored_results.append(result)
+
+        # Sort by total score (descending)
+        return await self._sort_results(scored_results, sort_key="score")
+
+    def __str__(self) -> str:
+        criteria_names = [c.name for c in self.config.scoring_criteria]
+        return f"{self.__class__.__name__} | criteria={criteria_names}"
+
+
 def create_reranker(
     reranker_type: RerankerType,
-    config: RerankerToolConfig | None = None,
+    config: RerankerToolConfig | LLMRerankerToolConfig | None = None,
```

Comment:
Technically don't need `LLMRerankerToolConfig` because it's also a type of `RerankerToolConfig`. Redundant type hint